### PR TITLE
Validate length of SSN

### PR DESCRIPTION
**Pull request by jessieay**
**WHY**: So we can know that it is 9 digits when entering it,
digit-by-digit, into the 1171 PDF.

Also adds auto-strip-attributes so that if someone has an extra
whitespace when entering their SSN it is auto-stripped

This approach was inspired by the SSN formatting in GCF, which I had a
hand in.

### CODE REVIEW

**Comment by jessieay** [comment]
[1171: Legibility](https://trello.com/c/bs9vQd1O/143-1171-legibility)

File: spec/support/shared_examples/social_security_number.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    it "does not allow dashes" do
+      subject.ssn = "  123-12-1234  "
+      expect(subject).to be_invalid
+      expect(subject.errors[:ssn]).to include(
+        "Make sure your SSN has 9 digits",
+      )
+    end
+  end
+end
```
Comment (jayroh) [inline spec/support/shared_examples/social_security_number.rb]: Does the auto_strip_attributes gem do the job of stripping out ALL non-integers before it's validated? 

If it doesn't do that - what do you think about masking the form input as a way to hold their hands in filling out the SSN? For many people it would ensure, in real-time, that they're entering it in the exact format we require.

Comment (jessieay) [inline spec/support/shared_examples/social_security_number.rb]: @jayroh no, it just strips whitespace. https://github.com/holli/auto_strip_attributes

the only thing I don't love about only allowing digits (I think that is what you are referring to when you say masking) is that it can sometimes feel unexpected. For example, our financial fields only allow numbers. Yesterday I was typing some words in there and nothing happened...so I thought my external keyboard was broken. Then I realized that numbers did work.

On the other hand, I can totally see why we'd want to help people not enter invalid info.

I think whether we want to do that is a UX question?

File: spec/controllers/personal_detail_controller_spec.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
       expect(step.sex).to eq("male")
       expect(step.marital_status).to eq("Married")
-      expect(step.ssn).to eq("12345")
+      expect(step.ssn).to eq("123456789")
```
Comment (adarsh) [inline spec/controllers/personal_detail_controller_spec.rb]: I would also add a test for SSNs which start with a zero. I have been bitten by this with Boston-area zipcodes (e.g. 02215).

Comment (jessieay) [inline spec/controllers/personal_detail_controller_spec.rb]: done at 412a648

File: spec/support/shared_examples/social_security_number.rb (hunk 1 above)
Comment (jayroh) [inline spec/support/shared_examples/social_security_number.rb]: Yeah, we can punt on this for now.

Comment (jayroh) [inline spec/support/shared_examples/social_security_number.rb]: As I was goign through the app on my phone, by the way, there were many many opportunities to make sure the phone KB was using the right keyboard set (numbers instead of letters, etc)

Comment (jessieay) [inline spec/support/shared_examples/social_security_number.rb]: created a card for that https://trello.com/c/8dWt7QB8
